### PR TITLE
Added RodCuttingProblem [JavaScript]

### LINKED ISSUE #354: Rod Cutting Problem [JavaScript]
#### This issue is about [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by Anshuman-Verma**
Fixes #354 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by Anshuman-Verma** [comment]
**DO NO REVIEW**

**Comment by Anshuman-Verma** [comment]
You can review now @aashutoshrathi

File: rod_cutting_problem/rodCuttingProblem.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+* Problem Statement: Given a rod of length n inches and an array of prices that contains prices of all pieces of size smaller than n.
```
Comment (Monal5031) [inline rod_cutting_problem/rodCuttingProblem.js]: mentioning only `inches` wouldn't be fair to other units. 😜

File: rod_cutting_problem/rodCuttingProblem.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+*/
+
+function rodCuttingProblem (price) {
+	/*
+	Computes maximum money that can be earned by cutting a rod of length n (Bottom-Up Approach).
+	Time Complexity	:	O(n ^ 2)
+	Space Complexity : O(n)
+	:param price:	Array in which price[i] denotes price	of rod of length i.
+	:return: returns optimal solution	for rod	of length n.
```
Comment (Monal5031) [inline rod_cutting_problem/rodCuttingProblem.js]: This line could be made better, as the solution is maximum.

File: rod_cutting_problem/rodCuttingProblem.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	*/
+	let n = price.length;
+	let bestPrice = new Array(n + 1);
+	bestPrice.fill(0);
+	for (let i = 1; i < bestPrice.length; i++) {
+		bestPrice[i] = price[i - 1];
+	}
+	for (let i = 1; i <= n; i++) {
+		for (let j = i; j <= n + 1; j++) {
```
Comment (Monal5031) [inline rod_cutting_problem/rodCuttingProblem.js]: are you sure `j <= n + 1`?

File: rod_cutting_problem/rodCuttingProblem.js (hunk 1 above)
Comment (Anshuman-Verma) [inline rod_cutting_problem/rodCuttingProblem.js]: Copied lol

Comment (Anshuman-Verma) [inline rod_cutting_problem/rodCuttingProblem.js]: Copied from py lol

**Comment by Anshuman-Verma** [comment]
Thanks  for review monty @Monal5031, take a look now ?

File: rod_cutting_problem/rodCuttingProblem.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	:return: returns maximum value obtainable by cutting up the rod and selling the pieces.
+	*/
+	let n = price.length;
+	let bestPrice = new Array(n + 1);
+	bestPrice.fill(0);
+	for (let i = 1; i < bestPrice.length; i++) {
+		bestPrice[i] = price[i - 1];
+	}
+	for (let i = 1; i <= n; i++) {
+		for (let j = i; j <= n; j++) {
```
Comment (aviaryan) [inline rod_cutting_problem/rodCuttingProblem.js]: It should be `j = 0; j < i;`. 🤔

Comment (Anshuman-Verma) [inline rod_cutting_problem/rodCuttingProblem.js]: The inner for loop content is different so it gives correct ans.
Take a look @aviaryan 
This is similar to the java implementation that we have.

Comment (aviaryan) [inline rod_cutting_problem/rodCuttingProblem.js]: It seems same to me as that GFG article. 
Plus this method is harder to understand if one follows DP principles. 
Please convert it to GFG style.

Comment (Anshuman-Verma) [inline rod_cutting_problem/rodCuttingProblem.js]: @aviaryan looks good to me now.
Does this mean Java implementation needs update too ?

**Pull request merged**